### PR TITLE
feat(linter): Implement eslint/no-object-constructor

### PR DESCRIPTION
**Pull request by azihsoyn**
Hello, again!

I added eslint rule eslint/no-object-constructor
https://eslint.org/docs/latest/rules/no-object-constructor

#### Notes
A test case for `new Object()` has been commented out:

This is due to the test configuration specifying `globals: { Object: "off" }`.
This approach follows the example set by the no_new_wrappers rule. [Reference Code](https://github.com/oxc-project/oxc/blob/bf839c1dfa1dcf6a48ec1fcadf49a1fe839e6f06/crates/oxc_linter/src/rules/eslint/no_new_wrappers.rs#L88)

### CODE REVIEW

**Review by no-yan — COMMENTED** [review]
Thank you for your contribution!
Since the team seem to be focusing on other tasks, I’ve added comments on smaller points to help keep things moving efficiently.

File: crates/oxc_linter/src/rules/eslint/no_object_constructor.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    ];
+
+    let fail = vec![
+        "new Object",
+        "Object()",
+        "const fn = () => Object();",
+        "Object() instanceof Object;",
+        "const obj = Object?.();",
+        "(new Object() instanceof Object);",
+    ];
```
[the remark this answers is not in the document]
Comment (azihsoyn) [reply crates/oxc_linter/src/rules/eslint/no_object_constructor.rs:102]: Thanks!

File: crates/oxc_linter/src/rules/eslint/no_object_constructor.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            return;
+        };
+
+        if ident.is_global_reference_name("Object", ctx.symbols())
+            && arguments.len() == 0
+            && type_parameters.is_none()
+        {
+            ctx.diagnostic(
+                crate::rules::eslint::no_object_constructor::no_object_constructor_diagnostic(span),
+            );
```
[the remark this answers is not in the document]
Comment (azihsoyn) [reply crates/oxc_linter/src/rules/eslint/no_object_constructor.rs:76]: Thanks!

File: crates/oxc_linter/src/rules/eslint/no_object_constructor.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (azihsoyn) [reply crates/oxc_linter/src/rules/eslint/no_object_constructor.rs:191]: like this?
[c4f2291](https://github.com/oxc-project/oxc/pull/7345/commits/c4f229177e8550c74a4feab0f0d30c402ef1f553)

**Review by camc314 — APPROVED** [review]
thank you! apologies for the delay reviewing!

**Pull request merged**